types/next/router.d.ts

```diff
@@ -20,7 +23,7 @@ export interface SingletonRouter {
     // router methods
     reload(route: string): Promise<void>;
     back(): void;
-    push(url: string, as?: string, options?: EventChangeOptions): Promise<boolean>;
+    push(url: string|UrlLike, as?: string, options?: EventChangeOptions): Promise<boolean>;
     replace(url: string, as?: string, options?: EventChangeOptions): Promise<boolean>;
     prefetch(url: string): Promise<React.ComponentType<any>>;
```

Comment:
The relevant bits of code appear to be https://github.com/zeit/next.js/blob/f7d2b96d283edb3903dfaa28d5491dd3988c45d7/lib/router/router.js#L111-L123

So it looks like `replace` also takes the multi-typed `url` parameter.  
And for both `push` and `replace`, `as?` can be a `UrlLike` well.

Can you make those updates?  Otherwise, this looks good to me!  Thanks!

Reply:
No dramas 👍